source/grid/manifold.cc

```diff
@@ -62,6 +62,128 @@ get_new_point (const Quadrature<spacedim> &quad) const
 }
 
 
+
+template <>
+Tensor<1,2>
+Manifold<2, 2>::
+normal_vector (const typename Triangulation<2, 2>::face_iterator &face,
+               const Point<2> &p) const
+{
+  const int spacedim=2, dim=2;
+
+  Tensor<1,spacedim> tangent = ((p-face->vertex(0)).norm_square() > (p-face->vertex(1)).norm_square() ?
+                                get_tangent_vector(p, face->vertex(0)) :
+                                -get_tangent_vector(p, face->vertex(1)));
+  return cross_product_2d(tangent);
```

Comment:
Oh, I see. I forgot. Maybe add a comment like "Return a vector normal to the tangent; as stated in the documentation, it need not be normalized."

Same in the other location below.